### PR TITLE
Enhancement: Player SFX

### PR DESCRIPTION
**Pull request by wattsit**
### Framework for player sfx, I'm sure the code can be made much more effecient and clean. 

**Key to note that the sounds are assigned to audio buses which can be accessed at the bottom.** 
**From here you can adjust volume and add effects as you please.**

image

### CODE REVIEW

File: Moonshot/player/PlayerAudio.gd [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		
+	elif sound == "sizzle":
+		pass
+		
+	elif sound == "heatlh":
+		pass
+	
+	elif sound == "pickup":
+		pass
+
```
Comment (wattsit) [inline Moonshot/player/PlayerAudio.gd:104]: I hate this

File: Moonshot/player/PlayerAudio.gd [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	elif sound == "death":
+		death_streamer.play()
+		
+	elif sound == "sizzle":
+		sizzle_streamer.play()
+		
+	elif sound == "health":
+		health_streamer.play()
+	
+	elif sound == "pickup":
```
Comment (whitew1994WW) [inline Moonshot/player/PlayerAudio.gd:73]: Could use a match here, would be a little neater

File: Moonshot/player/PlayerAudio.gd [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+onready var pain_streamer = create_streamer(pain_sfx)
+var death_sfx = load("res://player/assets/sounds/deathgrunt.wav")
+onready var death_streamer = create_streamer(death_sfx)
+var sizzle_sfx = load("res://player/assets/sounds/sizzle.wav")
+onready var sizzle_streamer = create_streamer(sizzle_sfx)
+var health_sfx = load("res://player/assets/sounds/healthpickup.wav")
+onready var health_streamer = create_streamer(health_sfx)
+var pickup_sfx = load("res://player/assets/sounds/game-pick-up-object.wav") 
+onready var pickup_streamer = create_streamer(pickup_sfx)
+
```
Comment (whitew1994WW) [inline Moonshot/player/PlayerAudio.gd:20]: Ok at the moment, but probably need a better way of organising these if we are gonna have any more sound effects.

File: Moonshot/player/PlayerAudio.gd (hunk 1 above)
Comment (whitew1994WW) [reply Moonshot/player/PlayerAudio.gd:104]: Haha, it looks like the audio functionality is really awkward to use

File: Moonshot/player/PlayerAudio.gd [hunk 4]
```diff
@@ -0,0 +1,105 @@
+extends Node
```
Comment (whitew1994WW) [inline Moonshot/player/PlayerAudio.gd:1]: Would it make more sense to have sound effects managed by an autoloader? I'm not sure just wanted to know if there was a reason that you chose to do it as a child of player.

File: Moonshot/player/player.gd [hunk 5]
```diff
@@ -157,24 +160,33 @@ func set_invulnerable(time : float, animation_name = "stagger") -> void:
...
+			return false
+
+func damagetile_impulse():
+	$SFX.play("sizzle")
+	if is_on_floor():
+		take_damage(0.1,Vector2.UP, true)
+		
+	elif is_on_wall():
+		var wall_normal = $LedgeWallDetector.scale.x * - 1
+		take_damage(0.1,Vector2(wall_normal,0) , true)
```
Comment (whitew1994WW) [inline Moonshot/player/player.gd:181]: Ah is this a fix for the damage tiles as well? 😄

**Review by whitew1994WW — COMMENTED** [review]
Thanks for doing this Bill, looks really good. I think my main comment is that this should probably be in an autoloader so that all objects can control SFX. It sounds 🐳  I am also getting this error: 
image

**Comment by wattsit** [comment]
> Thanks for doing this Bill, looks really good. I think my main comment is that this should probably be in an autoloader so that all objects can control SFX. It sounds 🐳 I am also getting this error: IMG
@will-white-dativa 

Somehow the player.tscn didn't get included in the pull, I've now added it. https://github.com/kepler471/moonshot/pull/212/commits/7022ea08b7d21b519d7d66d0f01cf0f3ed8046c7

**Comment by wattsit** [comment]
Used match statements for the play and stop function. https://github.com/kepler471/moonshot/pull/212/commits/effaa0f6ab9bdc81515d94e8521dedce060fcbae

Much cleaner, thanks @will-white-dativa .

File: Moonshot/player/player.tscn [hunk 6]
```diff
@@ -120,21 +121,23 @@ animations = [ {
 "Player",
 ]]
 collision_layer = 2
-collision_mask = 269
+collision_mask = 333
```
Comment (wattsit) [inline Moonshot/player/player.tscn:124]: Fix for the Player collision mask, adds damagetiles bit.

File: Moonshot/player/player.gd [hunk 7]
```diff
@@ -32,6 +32,7 @@ var animation_name
 var playing_reverse
 var invulnerable = false
 var safety = false
+var dead = false
```
Comment (wattsit) [inline Moonshot/player/player.gd:35]: Death flag added to the player.

File: Moonshot/player/player.gd (hunk 5 above)
Comment (wattsit) [reply Moonshot/player/player.gd:181]: This is mostly just splitting the impule logic from the actual damagetile check. Allowed me to use the check somewhere else as well.

**Review by whitew1994WW — APPROVED** [review]
Yeah sounds good to me ;) Do we want the other guns to have the same sound as the main laser blaster?

**Comment by wattsit** [comment]
> Yeah sounds good to me ;) Do we want the other guns to have the same sound as the main laser blaster?

Probably not.

**Pull request merged**